### PR TITLE
Migrate code to 4.4

### PR DESCRIPTION
**Pull request by boubaker**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/spacesadministration/SpacesAdministrationStorage.java [hunk 1]
```diff
@@ -1,228 +1,125 @@
...
+import org.exoplatform.commons.api.settings.data.Scope;
+import org.exoplatform.services.security.MembershipEntry;
+
 /**
  * @author Thomas Delhoménie
  */
 public class SpacesAdministrationStorage {
 
-  private Log log = ExoLogger.getLogger(SpacesAdministrationStorage.class);
+  public final static String SETTINGS = "spacesadministrationsettings";
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/addons/spacesadministration/SpacesAdministrationStorage.java]: No need to add the "settings" suffix IMO.

File: services/src/main/java/org/exoplatform/addons/spacesadministration/SpacesAdministrationStorage.java [hunk 2]
```diff
@@ -1,228 +1,125 @@
...
-      settingsNode.save();
-    } finally {
-      if (session != null) {
-        session.logout();
+    if (settingValue == null) {
+      settingValue = SettingValue.create(membershipToAdd);
+    } else {
+      String values = (String) settingValue.getValue();
+      if (!values.contains(membershipToAdd)) {
+        values += ";" + membershipToAdd;
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/addons/spacesadministration/SpacesAdministrationStorage.java]: Is it intended to change the separator ?

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]
@rdenarie @boubaker I added some little comments, but the main thing missing here is an upgrade plugin to move data from JCR to new Settings storage IMO. Do you think it is worth it ?

**Comment by thomasdelhomenie** [comment]
I updated the PR by considering the feedbacks and adding a migration service to move data from JCR to new Settings AI.

**Comment by rdenarie** [comment]
Ok for me. Tested from 4.4.3 to 4.4.4

**Comment by thomasdelhomenie** [comment]
OK thanks. I will merge the PR.

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**